blog/2022-11-28-ciam-101-authn-identity-sso/index.mdx

```diff
@@ -0,0 +1,362 @@
+---
+slug: ciam-101-intro-authn-sso
+authors: gao
+tags: [ciam, auth, authentication]
+---
+
+# CIAM 101: Authentication, Identity, SSO
+
+## Background
+
+I began the build Logto because I noticed that Identity and Access Management (IAM) had become increasingly complex and expansive over time. The concept of IAM is even large enough to give rise to new concepts, such as WIAM (Workforce IAM) and CIAM (Customer IAM).
+
+While WIAM and CIAM share the same foundation, they have distinct use cases: WIAM is typically used for internal users, while CIAM is used for external customers.<!--truncate--> Some examples:
+
+- **WIAM** Your company has a unified identity system for employees, thus every one can use the same account to access company resources, such as software subscriptions, cloud computing services, etc.
+- **CIAM** Your online bookstore requires a user identity system for customers and sellers. The sign-in experience is a critical part of onboarding, as it is located at the top of the conversion funnel.
+
+Logto started with the CIAM for various reasons (we’ll have another article talking about this). During development, we realized that building a unified understanding across the team would be beneficial before taking our product to the next level. We hope this will also help you gain a better grasp of the IAM landscape.
+
+Let’s get started!
+
+## The basics of CIAM
+
+:::info
+Regardless of their differences, WIAM and CIAM both have the same foundation: authentication and authorization. These two concepts are at the core of their features.
+:::
+
+In this article, we'll focus on the fundamental concepts of CIAM and problems we may meet during or after the authentication flow. We’ll also discuss Single Sign-On (SSO) and its related scenarios.
+
+### Authentication and authorization
+
+:::info Definition
+- Authentication (AuthN) answers the question “Who are you?”
+- Authorization (AuthZ) answers the question “What can you do?”
+:::
+
+If you discover something that does not fit into either of these two categories, it is likely not essential to the identity business.
+
+- **Examples for authentication**
+  - Password sign-in, passwordless sign-in, social sign-in, etc.
+  - Machine-to-Machine authentication
+- **Examples for authorization**
+  - Role-based Access Control
+  - Attribute-based Access Control
+- **Examples for exceptions**
+  - Non-identity data
+  - Web hooks
+
+### Identity and Tenant
+
+Identity typically represents either a user or a machine. Upon successful authentication, an ID Token is issued as an Identity.
+
+In other words, the main purpose of authentication is to obtain an Identity.
+
+A Tenant is a group of identities:
```

Comment:
```suggestion
A tenant is a group of identities:
```

Reply:
this is by design since we use "Tenant" as a term